node/full_node_integration_test.go

```diff
@@ -169,68 +169,120 @@ func TestLazyAggregator(t *testing.T) {
 	assert.False(node.IsRunning())
 	assert.NoError(err)
-	err = node.Start()
-	assert.NoError(err)
+
+	assert.NoError(node.Start())
 	defer func() {
 		require.NoError(node.Stop())
 	}()
 	assert.True(node.IsRunning())
 
-	require.NoError(err)
-
-	require.NoError(waitForFirstBlock(node.(*FullNode), false))
+	require.NoError(waitForFirstBlock(node.(*FullNode), Header))
 
 	client := node.GetClient()
 
 	_, err = client.BroadcastTxCommit(context.Background(), []byte{0, 0, 0, 1})
 	assert.NoError(err)
-	require.NoError(waitForAtLeastNBlocks(node, 2, false))
+	require.NoError(waitForAtLeastNBlocks(node, 2, Header))
 
 	_, err = client.BroadcastTxCommit(context.Background(), []byte{0, 0, 0, 2})
 	assert.NoError(err)
-	require.NoError(waitForAtLeastNBlocks(node, 3, false))
+	require.NoError(waitForAtLeastNBlocks(node, 3, Header))
 
 	_, err = client.BroadcastTxCommit(context.Background(), []byte{0, 0, 0, 3})
 	assert.NoError(err)
 
-	require.NoError(waitForAtLeastNBlocks(node, 4, false))
+	require.NoError(waitForAtLeastNBlocks(node, 4, Header))
+}
+
+// TestSingleAggregatorTwoFullNodesBlockSyncSpeed tests the scenario where the chain's block time is much faster than the DA's block time. In this case, the full nodes should be able to use block sync to sync blocks much faster than syncing from the DA layer, and the test should conclude within block time
+func TestSingleAggregatorTwoFullNodesBlockSyncSpeed(t *testing.T) {
+	require := require.New(t)
+	aggCtx, aggCancel := context.WithCancel(context.Background())
+	defer aggCancel()
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	clientNodes := 3
+	bmConfig := getBMConfig()
+	bmConfig.BlockTime = 1 * time.Second
+	bmConfig.DABlockTime = 10 * time.Second
+	const numberOfBlocksTSyncTill = 5
+
+	ch := make(chan struct{})
+	timer := time.NewTimer(numberOfBlocksTSyncTill * bmConfig.BlockTime)
+
+	go func() {
+		select {
+		case <-ch:
+			// Channel closed before timer expired.
+			return
+		case <-timer.C:
+			// Timer expired before channel closed.
+			t.Error("nodes did not sync before DA Block time")
+			return
+		}
+	}()
+	nodes, _ := createNodes(aggCtx, ctx, clientNodes, bmConfig, t)
+
+	node1 := nodes[0]
+	node2 := nodes[1]
+	node3 := nodes[2]
 
+	require.NoError(node1.Start())
+	defer func() {
+		require.NoError(node1.Stop())
+	}()
+	require.NoError(waitForFirstBlock(node1, Store))
+	require.NoError(node2.Start())
+	defer func() {
+		require.NoError(node2.Stop())
+	}()
+	require.NoError(node3.Start())
+	defer func() {
+		require.NoError(node3.Stop())
+	}()
+
+	require.NoError(waitForAtLeastNBlocks(node2, numberOfBlocksTSyncTill, Store))
+	require.NoError(waitForAtLeastNBlocks(node3, numberOfBlocksTSyncTill, Store))
+
+	require.NoError(verifyNodesSynced(node1, node2, Store))
+	require.NoError(verifyNodesSynced(node1, node3, Store))
+	close(ch)
```

Comment:
learning from our previous mistakes, I realize this probably should be in a defer statement.
```go
ch := make(chan struct{})
defer close(ch)
```